lib/rgeo/cartesian/feature_methods.rb

```diff
@@ -50,26 +56,10 @@ def segments
       end
 
       def simple?
-        len = segments.length
-        return false if segments.any?(&:degenerate?)
-        return true if len == 1
-        return segments[0].s != segments[1].e if len == 2
-        segments.each_with_index do |seg, index|
-          nindex = index + 1
-          nindex = nil if nindex == len
-          return false if nindex && seg.contains_point?(segments[nindex].e)
-          pindex = index - 1
-          pindex = nil if pindex < 0
-          return false if pindex && seg.contains_point?(segments[pindex].s)
-          next unless nindex
-          oindex = nindex + 1
-          while oindex < len
-            oseg = segments[oindex]
-            return false if !(index == 0 && oindex == len - 1 && seg.s == oseg.e) && seg.intersects_segment?(oseg)
-            oindex += 1
-          end
-        end
-        true
+        # TODO: should we replace with graph.incident_edges.length == segments.length
```

Comment:
@keithdoggett do you want to discuss that one?

Reply:
Yeah so when the `GeometryGraph` is instantiated, it uses a `SweeplineIntersector` to figure out intersections and help with noding of the graph. So the idea is that calling `graph.incident_edges.length == segments.length` is equivalent to running the `SweeplineIntersector` and checking if it's empty.

That being said, creating the `GeometryGraph` adds overhead since, in addition to finding the intersections, it also creates the graph of `HalfEdges`. `graph` is memoized, so if it's used multiple times it makes sense IMO to use graph here, but if `graph` is only used to determine simplicity, I think the reduced overhead of `SweeplineIntersector` makes sense.

Let me know if that's clear.

Reply:
Left a note about it.